src/test_l2_dsDisplay.c

```diff
@@ -1,121 +1,172 @@
 */
 
 /**
- * @addtogroup HPK Hardware Porting Kit
- * @{
- * @par The Hardware Porting Kit
- * HPK is the next evolution of the well-defined Hardware Abstraction Layer
- * (HAL), but augmented with more comprehensive documentation and test suites
- * that OEM or SOC vendors can use to self-certify their ports before taking
- * them to RDKM for validation or to an operator for final integration and
- * deployment. The Hardware Porting Kit effectively enables an OEM and/or SOC
- * vendor to self-certify their own Video Accelerator devices, with minimal RDKM
- * assistance.
- *
- */
-
-/**
- * @addtogroup Device_Settings Device Settings Module
- * @{
- */
-
-/**
- * @addtogroup Device_Settings_HALTEST Device Settings HAL Tests
- * @{
- */
-
-/**
- * @addtogroup DS_Display_HALTEST Device Settings Display HAL Tests
- * @{
- */
-
-/**
- * @defgroup DS_Display_HALTEST_L2 Device Settings Display HAL Tests L2 File
- * @{
- * @parblock
- *
- * ### L2 Tests for DS Display HAL :
- *
- * Level 2 unit test cases for all APIs of Device Settings Display HAL
- *
- * **Pre-Conditions:**  None@n
- * **Dependencies:** None@n
- *
- * TODO: Refer to L2 Specification documentation : [l2_module_test_specification_template.md](../docs/pages/l2_module_test_specification_template.md)
- *
- * @endparblock
- */
+* @file test_l2_dsDisplay.c
+* @page dsDisplay Level 2 Tests
+*
+* ## Module's Role
+* This module includes Level 2 functional tests (success and failure scenarios).
+* This is to ensure that the dsDisplay APIs meet the requirements across all vendors.
+*
+* **Pre-Conditions:**  None@n
+* **Dependencies:** None@n
+*
+* Ref to API Definition specification documentation : [halSpec.md](../../../docs/halSpec.md)
+*/
 
+#include <ut.h>
+#include <ut_log.h>
+#include <stdlib.h>
+#include "dsDisplay.h"
 
 /**
- * @file test_l2_dsDisplay.c
- *
- */
-
-#include <string.h>
-#include <stdlib.h>
+* @brief This test validates the EDID of a display
+*
+* This test initializes the display, gets the display handle, retrieves the EDID and EDID bytes, and then terminates the display. The obtained EDID and EDID bytes are expected to be validated with predefined values.
+*
+* **Test Group ID:** 02@n
+* **Test Case ID:** 001@n
+*
+* **Test Procedure:**
+* Refer to UT specification documentation [l2_dsDisplay_test_specification.md](l2_dsDisplay_test_specification.md)
+*/
 
-#include <ut.h>
-#include <ut_log.h>
+void test_l2_dsDisplay_ValidateEDID(void)
+{
+    UT_LOG("Entering test_l2_dsDisplay_ValidateEDID...");
+
+    dsError_t ret;
+    intptr_t handle;
+    dsDisplayEDID_t *edid = (dsDisplayEDID_t*)malloc(sizeof(dsDisplayEDID_t));
+    unsigned char *edidBuffer = (unsigned char *)malloc(1024);
+    int length;
+
+    // Step 1
+    ret = dsDisplayInit();
+    UT_LOG("Invoked dsDisplayInit(). Return status: %d", ret);
+    UT_ASSERT_EQUAL(ret, dsERR_NONE);
+
+    if (ret == dsERR_NONE)
+    {
+        // Step 2
+        ret = dsGetDisplay(dsVIDEOPORT_TYPE_HDMI, 0, &handle);
```

Comment:
for sink devices display should be dsVIDEOPORT_TYPE_INTERNAL. dsVIDEOPORT_TYPE_HDMI is not supported. Spec needs to be updated with details

Reply:
Done